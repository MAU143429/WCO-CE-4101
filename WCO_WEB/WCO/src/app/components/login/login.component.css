body{
  height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 10px;
  background-color: white;
  background-image: url("../../../assets/images/bg.jpg");


}
.container{
  max-width: 700px;
  height: 650px;
  width: 100%;
  /**background-color: rgba(255, 255, 255, 0.295);*/
  background-color: white;
  padding: 25px 30px;
  border-radius: 5px;
  box-shadow: 0 5px 10px rgba(0,0,0,0.20);
}
.container .title{
  font-size: 35px;
  font-weight: 500;
  text-align: center;
  position: relative;
  margin-bottom: 2%;
}

.subtitle{
  font-size: 25px;
  font-weight: 500;
  text-align: center;
}

.container .title::before{
  content: "";
  position: absolute;
  bottom: 0;
  height: 3px;
  width: 210px;
  border-radius: 5px;
}
.content form .account-details{
  display: flex;
  flex-wrap: wrap;
  margin: 20px 0 12px 0;
}

form .account-details .input-box{
  margin-bottom: 40px;
  margin-left: 25%;
  width: calc(100% / 2);

}


form .input-box span.details{
  display: block;
  font-weight: 500;
  margin-bottom: 5px;
  text-align: center;
}



.account-details .input-box input{
  height: 45px;
  width: 100%;
  font-size: 16px;
  border-radius: 5px;
  padding-left: 15px;
  border: 1px solid #ccc;
  border-bottom-width: 2px;
  transition: all 0.3s ease;
}

form .button{
  height: 45px;
  margin: 35px 0
}

 /* Aspecto del boton de crear partido */

 form .button input{
  height: 100%;
  width: 30%;
  margin-left: 35%;
  border-radius: 5px;
  border: none;
  color: #fff;
  font-size: 18px;
  font-weight: 500;
  letter-spacing: 1px;
  cursor: pointer;
  transition: all 0.3s ease;
  background: linear-gradient(135deg, #1277D9, #7bb0e6, #1277D9);
}
 form .button input:hover{
  background: linear-gradient(135deg, #1277D9, #7bb0e6, #1277D9);
  }
 @media(max-width: 584px){
 .container{
  max-width: 100%;
}
}

.social
{
  position: absolute;
  z-index: 10;
  bottom: 20px;
  display: flex;
  justify-content: center;
  align-items: center;
}
.social li
{
  list-style: none;
}
.social li a
{
  display: inline-block;
  margin-right: 20px;
  filter: invert(1);
  transform: scale(0.5);
  transition: 0.5s;
}
.social li a:hover
{
  transform: scale(0.5) translateY(-15px);
}

@media(max-width: 459px){
  .container .content{
    flex-direction: column;
  }
}
