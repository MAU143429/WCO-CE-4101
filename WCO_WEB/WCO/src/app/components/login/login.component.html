<p><!DOCTYPE html>
  <html lang="en" dir="ltr">
    <head>
      <meta charset="UTF-8">
      <!---Referencia al archivo css que contiene el estilo--->
      <link rel="stylesheet" href="login.component.css">
       <meta name="viewport" content="width=device-width, initial-scale=1.0">
     </head>
  <body>
    <div class="container">
      <div class="title">Bienvenido a World Cup Online</div>
      <div class="content">
        <div class="subtitle">Iniciar Sesión</div>
        <br>
        <br>
        <form>
          <div class="account-details">
            <div class="input-box">
              <span class="details">Correo Electrónico<span style="color: rgb(245, 59, 59);">*</span> </span>
              <input type="email" minlength="5" maxlength="30" placeholder="Ingrese un correo electrónico.." required>
            </div>

            <div class="input-box">
              <span class="details">Contraseña<span style="color: rgb(245, 59, 59);">*</span> </span>
              <input type="password" minlength="6" maxlength="8" placeholder="Ingrese una contraseña aquí..." required>
            </div>
            <div class="input-box">
              <span >No tienes una cuenta aún? Regístrate <a routerLink="/register" style="color: rgb(4, 0, 255);">aquí</a></span>
            </div>
          </div>
          <div class="button">
            <input type="submit" value="Iniciar Sesión"routerLink="/home">
          </div>

        </form>
      </div>
    </div>
  </body>
  </html>
