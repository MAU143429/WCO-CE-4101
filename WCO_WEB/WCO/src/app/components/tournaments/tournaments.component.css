body{
  height: 100vh;
  justify-content: center;
  align-items: center;
  padding: 10px;
  background-color: white;

}
.container{
  max-width: 1700px;
  width: 100%;
  background-color: rgba(255, 255, 255, 0.295);
  padding: 25px 30px;
  border-radius: 5px;
  box-shadow: 0 5px 10px rgba(0,0,0,0.20);
}


.input-box{
  display: flex;
  margin-bottom: 15px;
  margin-left: 17%;
  width: 70%;
  font-size: 20px;
}

.title{
  width: 40%;
  font-size: 30px;
  font-weight: 500;
  margin-top: 1%;
  position: relative;
  justify-content: space-between;

}

.input-box input{
  height: 45px;
  width: 100%;
  font-size: 16px;
  border-radius: 5px;
  padding-left: 15px;
  border: 1px solid #ccc;
  border-bottom-width: 2px;
  transition: all 0.3s ease;
}


