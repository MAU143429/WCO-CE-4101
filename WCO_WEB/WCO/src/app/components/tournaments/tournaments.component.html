<!DOCTYPE html>
<html lang="en" dir="ltr">
  <head>
    <meta charset="UTF-8">
    <!---Referencia al archivo css que contiene el estilo--->
    <link rel="stylesheet" href="tournaments.component.css">
     <meta name="viewport" content="width=device-width, initial-scale=1.0">
   </head>
<!---Componentes que conforman la seccion de ver todos los torneos--->
<app-navbar></app-navbar>
<div class="input-box">
  <div class="title">Todos los torneos</div>
  <input  type="submit" value="Crear Torneo" routerLink="/create-tournaments">
</div>
<body>
  <div class="container">
    <div class="content">

      <table id="tournaments" class="table table-bordered mt-4 mytable">
        <tr class="table-primary mytable">
          <th>Nombre del Torneo</th>
          <th>ID</th>
          <th>Fecha de inicio</th>
          <th>Fecha de final</th>
          <th>Detalles</th>
        </tr>
      <tr *ngFor="let data of tournamentsData">
        <td>{{data.name}}</td>
          <td>{{data.toId}}</td>
          <td>{{data.startDate}}</td>
          <td>{{data.endDate}}</td>
          <td><button class="viewbtn" routerLink="/tournament-details" (click)="openTournament(data.toId)" >Ver Torneo</button></td>
      </tr>
      </table>
    </div>
  </div>
</body>
</html>


