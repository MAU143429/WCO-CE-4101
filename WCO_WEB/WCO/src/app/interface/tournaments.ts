import { DbBracket } from '../model/db-bracket';
import { DbTeam } from '../model/db-team';

export interface Tournaments {
  toId: string;
  name: string;
  startDate: string;
  endDate: string;
  type: string;
  description: string;
  teams: DbTeam[];
  brackets: DbBracket[];
}
