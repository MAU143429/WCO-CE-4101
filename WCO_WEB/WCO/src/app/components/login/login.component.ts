import { Component, OnInit } from '@angular/core';
import { NgbDateNativeAdapter } from '@ng-bootstrap/ng-bootstrap';
import { Login } from 'src/app/model/login';
import { CreateAccount } from '../../model/create-account';
import { AccountService } from '../../services/account.service';
import {MatSnackBar} from '@angular/material/snack-bar';
import { Router } from '@angular/router';


@Component({
  selector: 'app-login',
  templateUrl: './login.component.html',
  styleUrls: ['./login.component.css'],
})
export class LoginComponent implements OnInit {
  Users: CreateAccount[] = [];
  newLogin: Login = new Login();

  /**
   * Este metodo permite realizar un pequeño delay
   * @param ms el tiempo del delay en ms
   */
   async delay(ms: number) {
    await new Promise<void>((resolve) => setTimeout(() => resolve(), ms)).then(
      () => console.log('fired')
    );
  }  

  constructor(
    private service: AccountService,
    private _snackBar: MatSnackBar,
    private router: Router
    ) {}

  ngOnInit(): void {}

  /**
   * Metodo para mostrar alerta de error por 2 segundos
   * @param message1 Mensaje de error
   * @param message2 Mensaje para cerrar alerta
   */
  openError(message: string) {
    this._snackBar.open(message, 'Intente de nuevo',{
      duration: 2000,
      horizontalPosition: 'center',
      verticalPosition: 'top',
      panelClass: 'red-snackbar',      
    })
  };

  /**
   * Metodo para mostrar alerta de exito por 2 segundos
   * @param message1 Mensaje de exito
   * @param message2 Mensaje para cerrar alerta
   */
  openSuccess(message1: string, message2: string) {
    this._snackBar.open(message1, message2, {
      duration: 2000,
      horizontalPosition: 'center',
      verticalPosition: 'top',
      panelClass: 'green-snackbar',
    });
  }


  /**
   * Este metodo permite hacer todas las verificaciones del inicio de sesion
   * asi como la comprobacion de contraseñas y posteriormente el paso a WCO
   */
  addLogin() {

    this.delay(50).then(() => {
      this.service
        .getAccountByEmail(this.newLogin.email)
        .subscribe((data) => (this.Users = data));
    });

    //Verificar que esten todos los espacios requeridos
    if ((this.newLogin.email == "" ||
         this.newLogin.password == "")){
      this.openError("Faltan espacios requeridos para iniciar sesión")
    }
    // Verificar es que el correo tenga una cuenta en WCO (Si verifico esto, y despues digo que credenciales estan mal, entonces es obvio que la contraseña es la que está mal)
    else if (!(this.Users.length > 0)){
      this.openError("Cuenta con correo no registrado")
    }
    // Verificar las credenciales
    else if (!(this.Users[0].password == this.newLogin.password)){
      this.openError("Credenciales incorrectos")
    } else {
      this.openSuccess('Cuenta creada con éxito', 'Ok');
      this.router.navigate(['/home']);
    }
  }
}
