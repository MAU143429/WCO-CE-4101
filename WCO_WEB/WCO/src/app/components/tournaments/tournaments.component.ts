import { Component, Input, OnInit } from '@angular/core';
import { TournamentService } from '../../services/tournament.service';
import { InternalService } from '../../services/internal.service';
import { Tournaments } from '../../interface/tournaments';

@Component({
  selector: 'app-tournaments',
  templateUrl: './tournaments.component.html',
  styleUrls: ['./tournaments.component.css'],
})
export class TournamentsComponent implements OnInit {
  tournamentsData: Tournaments[] = [];
  constructor(
    private service: TournamentService,
    private connection: InternalService
  ) {}

  ngOnInit(): void {
    this.service.getTournaments().subscribe((data: Tournaments[]) => {
      this.tournamentsData = data;
    });
  }

  openTournament(id: string) {
    this.connection.setTournamentId(id);
    //this.connection.sendTournamentKey(id);
  }
}
