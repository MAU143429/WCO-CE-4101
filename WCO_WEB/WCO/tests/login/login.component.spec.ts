import { ComponentFixture, fakeAsync, TestBed, tick } from '@angular/core/testing';
import { HttpClientTestingModule, HttpTestingController } from '@angular/common/http/testing'

import { AccountService } from '../../src/app/services/account.service';
import { LoginComponent } from '../../src/app/components/login/login.component';
import { TournamentsComponent } from '../../src/app/components/tournaments/tournaments.component'
import { MatSnackBar, MatSnackBarModule } from '@angular/material/snack-bar';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { EventManager } from '@angular/platform-browser';
import { BrowserAnimationsModule, NoopAnimationsModule } from '@angular/platform-browser/animations';
import { Router } from '@angular/router';
import { RouterTestingModule } from '@angular/router/testing';
import { Observable, observable, of } from 'rxjs';

const dummyAccount = [{
  email: 'prueba@gmail.com',
  password: 'lunes233' 
}];

const mockAccountService: {
  createAccount: () => Observable<any>,
  getAccountByEmail: () => Observable<any>,
  newLogin: () => Observable<any>,
  getRole: () => Observable<any>
} = {
  createAccount: () => of(dummyAccount),
  getAccountByEmail: () => of([{
    name: 'David',
    lastname: 'Soto',
    email: 'prueba@gmail.com',
    password: 'lunes233',
    nickname: 'dasotovid',
    birthdate: '01/01/01',
    country: 'Chile'
  }]),
  newLogin: () => of(dummyAccount),
  getRole: () => of(true)
};

describe('LoginComponent', () => {
  let component: LoginComponent;
  let fixture: ComponentFixture<LoginComponent>;
  let compiled: HTMLElement;
  let service: AccountService;
  let httpMock: HttpTestingController;
  let router: Router;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [ LoginComponent ],
      imports: [ 
        HttpClientTestingModule, 
        MatSnackBarModule, 
        FormsModule, 
        ReactiveFormsModule,
        BrowserAnimationsModule,
        NoopAnimationsModule,
        RouterTestingModule.withRoutes([{path: 'home', component: TournamentsComponent}])
        ],
      providers: [ 
        { provide: AccountService, useValue: mockAccountService }
      ]
    })
    .compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(LoginComponent);
    component = fixture.componentInstance;
    service = TestBed.inject( AccountService );
    httpMock = TestBed.inject( HttpTestingController );
    router = TestBed.inject(Router);

    fixture.detectChanges();
    compiled = fixture.nativeElement;
  });

  test('should create', () => {
    expect(component).toBeTruthy();
  });

  test('debe de hace match con el snapshot', () => {
    expect(compiled.innerHTML).toMatchSnapshot();
  });

  test('se llama addLogin() al presionar boton de ingresar', () => {
    const btnLogin = compiled.querySelector('[data-test=btn-login]');
    const spy = jest.spyOn( component, 'addLogin') ;
    btnLogin?.dispatchEvent( new Event( 'click' ) );
    console.log(btnLogin);
    expect(spy).toHaveBeenCalled();
  });

  test('sirve input de login', () => {
    const emailInput: HTMLInputElement = fixture.debugElement.nativeElement.querySelector('#emailInput');
    emailInput.value = 'prueba@gmail.com'
    emailInput.dispatchEvent(new Event('input'));

    const passwordInput: HTMLInputElement = fixture.debugElement.nativeElement.querySelector('#passwordInput');
    passwordInput.value = 'lunes233'
    passwordInput.dispatchEvent(new Event('input'));

    console.log("SIRVE ",component.newLogin.email, component.newLogin.password);

    expect( component.newLogin.email = emailInput.value);
    expect( component.newLogin.password = passwordInput.value);
    
  });

  test( 'faltan espacios requeridos', fakeAsync(() => {
    const spy = jest.spyOn( component, 'openError').mockImplementation();
    //const btnLogin = compiled.querySelector('[data-test=btn-login]');

    const emailInput: HTMLInputElement = fixture.debugElement.nativeElement.querySelector('#emailInput');
    emailInput.value = 'prueba@gmail.com'
    emailInput.dispatchEvent(new Event('input'));

    component.addLogin();
    tick(50);
    
    expect(spy).toHaveBeenCalled();
    
    
  }));
  
  test('deberia verificar datos y hacer route a home', fakeAsync(() => {

    const getAccSpy = jest.spyOn(service, "getAccountByEmail");
    getAccSpy.mockReturnValue(of([{
      name: 'David',
      lastname: 'Soto',
      email: 'prueba@gmail.com',
      password: 'lunes233',
      nickname: 'dasotovid',
      birthdate: '01/01/01',
      country: 'Chile',
    }]));

    const newLogInSpy = jest.spyOn(mockAccountService, "newLogin")
    newLogInSpy.mockReturnValue(of(dummyAccount));

    const getRoleSpy = jest.spyOn(mockAccountService, "getRole");
    getRoleSpy.mockReturnValue(of(true))

    const routeSpy = jest.spyOn(router, "navigate");


    const emailInput: HTMLInputElement = fixture.debugElement.nativeElement.querySelector('#emailInput');
    emailInput.value = 'prueba@gmail.com'
    emailInput.dispatchEvent(new Event('input'));

    const pwInput: HTMLInputElement = fixture.debugElement.nativeElement.querySelector('#passwordInput');
    pwInput.value = 'lunes233'
    pwInput.dispatchEvent(new Event('input'));

    component.addLogin();
    tick(800);

    console.log(component.Users[0])
    expect(routeSpy).toHaveBeenCalled();

    // revisar que se haga el route a donde sea que ocupe
  }));
  
});
