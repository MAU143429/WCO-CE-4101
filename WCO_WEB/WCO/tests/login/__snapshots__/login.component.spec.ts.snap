// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginComponent debe de hace match con el snapshot 1`] = `"<p><html lang=\\"en\\" dir=\\"ltr\\" ng-reflect-dir=\\"ltr\\"><head><meta charset=\\"UTF-8\\"><meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\"></head><body><div class=\\"container\\"><div class=\\"title\\">Bienvenido a World Cup Online</div><div class=\\"content\\"><div class=\\"subtitle\\">Iniciar Sesión</div><br><br><form novalidate=\\"\\" class=\\"ng-untouched ng-pristine ng-valid\\"><div class=\\"account-details\\"><div class=\\"input-box\\"><span class=\\"details\\">Correo Electrónico<span style=\\"color: rgb(245, 59, 59);\\">*</span></span><input type=\\"email\\" maxlength=\\"30\\" placeholder=\\"Ingrese un correo electrónico..\\" id=\\"emailInput\\" required=\\"\\" ng-reflect-required=\\"\\" ng-reflect-maxlength=\\"30\\" ng-reflect-model=\\"\\" ng-reflect-options=\\"[object Object]\\" class=\\"ng-untouched ng-pristine ng-invalid\\"></div><div class=\\"input-box\\"><span class=\\"details\\">Contraseña<span style=\\"color: rgb(245, 59, 59);\\">*</span></span><input type=\\"password\\" minlength=\\"6\\" maxlength=\\"8\\" placeholder=\\"Ingrese una contraseña aquí...\\" id=\\"passwordInput\\" required=\\"\\" ng-reflect-required=\\"\\" ng-reflect-minlength=\\"6\\" ng-reflect-maxlength=\\"8\\" ng-reflect-model=\\"\\" ng-reflect-options=\\"[object Object]\\" class=\\"ng-untouched ng-pristine ng-invalid\\"></div><div class=\\"input-box\\"><span>No tienes una cuenta aún? Regístrate <a routerlink=\\"/register\\" style=\\"color: rgb(4, 0, 255);\\" ng-reflect-router-link=\\"/register\\" href=\\"/register\\">aquí</a></span></div></div><div class=\\"button\\"><input type=\\"submit\\" value=\\"Iniciar Sesión\\" data-test=\\"btn-login\\"></div></form></div></div><ul class=\\"social\\"><li><a href=\\"https://www.facebook.com/fifaworldcup/\\"><img src=\\"https://i.ibb.co/x7P24fL/facebook.png\\"></a></li><li><a href=\\"https://twitter.com/fifaworldcup_es\\"><img src=\\"https://i.ibb.co/Wnxq2Nq/twitter.png\\"></a></li><li><a href=\\"https://www.instagram.com/fifaworldcup/?hl=es\\"><img src=\\"https://i.ibb.co/ySwtH4B/instagram.png\\"></a></li></ul></body></html></p>"`;
